mgmt/imgmgr/src/imgmgr.c

```diff
@@ -564,19 +606,22 @@ imgr_upload_inspect(const struct imgr_upload_req *req,
             return MGMT_ERR_ENOMEM;
         }
 
+#if MYNEWT_VAL(IMGMGR_LAZY_ERASE) == 0
         rc = flash_area_open(action->area_id, &fa);
         if (rc) {
             *errstr = imgmgr_err_str_flash_open_failed;
             return MGMT_ERR_EUNKNOWN;
         }
 
+        bool empty;
```

Comment:
variables should be declared at beginning of function. you can either `#ifdef` it there or alternatively use `(void)empty;` so compiler won't complain - I'm fine with either way of doing this.